File: spring-boot-postgres-java/src/test/java/de/larmic/postgres/database/TweetRepositoryTest.java
package de.larmic.postgres.database;

import de.larmic.postgres.database.model.TweetEntity;
import de.larmic.postgres.testcontainers.AbstractDatabaseTest;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.within;

/**
 * A fast unit tests to start only database context.
 */
class TweetRepositoryTest extends AbstractDatabaseTest {

    @Test
    void save() {
        final var entity = new TweetEntity("Some test message");

        this.tweetRepository.save(entity);

        assertThat(entity.getId()).isNotZero();
        assertThat(entity.getMessage()).isEqualTo("Some test message");
        assertThat(entity.getCreateDate()).isCloseTo(LocalDateTime.now(), within(1, ChronoUnit.SECONDS));
        assertThat(entity.getLastUpdateDate()).isCloseTo(LocalDateTime.now(), within(1, ChronoUnit.SECONDS));
    }
}

File: spring-boot-postgres-kotlin/src/test/kotlin/de/larmic/postgres/database/TweetRepositoryTest.kt
package de.larmic.postgres.database

import de.larmic.postgres.database.model.TweetEntity
import de.larmic.postgres.testcontainers.AbstractDatabaseTest
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit

class TweetRepositoryTest : AbstractDatabaseTest() {

    @Test
    fun save() {
        val entity = TweetEntity()
        entity.message = "Some test message"

        tweetRepository.save(entity)
        Assertions.assertThat(entity.id).isNotZero
        Assertions.assertThat(entity.message).isEqualTo("Some test message")
        Assertions.assertThat(entity.createDate).isCloseTo(LocalDateTime.now(), Assertions.within(1, ChronoUnit.SECONDS))
        Assertions.assertThat(entity.lastUpdateDate).isCloseTo(LocalDateTime.now(), Assertions.within(1, ChronoUnit.SECONDS))
    }
}